src/_pytest/_code/code.py

```diff
@@ -572,8 +572,9 @@ def match(self, regexp):
         raised.
         """
         __tracebackhide__ = True
-        if not re.search(regexp, str(self.value)):
-            assert 0, "Pattern '{!s}' not found in '{!s}'".format(regexp, self.value)
+        value = safe_str(self.value)
+        if not re.search(regexp, value):
+            assert 0, "Pattern '{!s}' not found in '{!s}'".format(regexp, value)
```

Comment:
option or followup: we might want to use `raise AssertionError(...)` in that case

Reply:
Done

Out of curiosity, is there any reason other than style to prefer one over the other?